tide-whisperer.go

```diff
@@ -167,6 +147,39 @@ func main() {
 		log.Printf("%s mongo processing finished after %.5f seconds and returned %d records", DATA_API_PREFIX, time.Now().Sub(startTime).Seconds(), writeCount)
 	}
 
+	authorize := func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			start := time.Now()
+
+			token := r.Header.Get("x-tidepool-session-token")
+			accessTokenHeader := r.Header.Get("Authorization")
+			if len(accessTokenHeader) > 7 && strings.EqualFold(accessTokenHeader[0:7], "BEARER ") {
+				log.Println("Validating with access_token")
+				token = strings.Split(accessTokenHeader, " ")[1]
+			}
+
+			if token != "" {
+				queryParams, err := store.GetParams(r.URL.Query(), &config.SchemaVersion)
+				if err != nil {
+					jsonError(w, error_invalid_parameters, start)
+					return
+				}
+
+				td := shorelineClient.CheckToken(token)
```

Comment:
I'd put this before the `store.GetParams`. First, check the token. If `td` is `nil`, then bail. Then get the `queryParams` and check against server/user id. Otherwise, we'll send back an empty response.